.github/workflows/update-sysdig-deploy-chart.yaml

```diff
@@ -0,0 +1,49 @@
+name: update_sysdig_deploy_chart_versions
+run-name: github actions is updating sysdig-deploy versions based on dependencies
+on:
+  push:
+    branches:
+      - master
+jobs:
+  update-sysdig-deploy-version:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Git checkout
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+      - name: Check if dependent charts were modified or not
+        uses: tj-actions/changed-files@v14.3
+        id: dependent_files
+        with:
+          files: |
+            charts/node-analyzer/*
+            charts/agent/*
+            charts/kspm-collector/*
+            charts/rapid-response/*
+            charts/admission-controller/*
+      - name: Install YQ
+        if: steps.dependent_files.outputs.any_changed == 'true'
+        uses: dcarbone/install-yq-action@v1.0.1
```

Comment:
I'm not particularly fond of those duplicates:
a) paths being duplicated on both the `.yaml` file AND the shell script
b) the gating condition `steps.dependent_files.outputs.any_changed == 'true'` all over the steps

For a) I would try to shift the responsibility onto the script so that it returns `0` when `No modifications necessary` and `1` in all other cases.
For b) it looks like an early-exit action would be the best solution, which is much-desired yet not available: https://github.com/actions/runner/issues/662.  A sub-optimal solution would be an environment variable as suggested here: https://github.com/actions/runner/issues/662#issuecomment-1295229779

Reply:
Hello for a) I can remove the paths from the workflow the only thing I can foresee is that it would run this workflow on any change made on this repository as a push to master, which should be fine since it won't make any changes. Let me know if that is preferable.
Regarding return 0 which means execution is a success is fine but if I return 1 shouldn't that lead to erroring out and the workflow not running successfully?
b) I am assuming what is meant here is to refer to an environment variable after setting it once instead of the whole path. ill make that change now.

Reply:
Hi @iurly I tried those changes and although it works fine when changes are present it still runs the subsequent changes when the changes are not present ("No modifications necessary"). I am more inclined towards keeping the changes currently present seeing that.